---
title: PartitionFinder2的一般使用方法（中文版）
author: Brandon Stark
date: '2018-04-09'
categories:
  - R
tags:
  - R Markdown
slug: partitionfinder2的一般使用方法（中文版）
---



<div id="section" class="section level2">
<h2>———————————————————————————————–</h2>
<table>
<tbody>
<tr class="odd">
<td align="left"><strong><font size = 16><a href="https://github.com/roblanf">Rob Lanfear</a></font></strong></td>
</tr>
<tr class="even">
<td align="left"><img src="http://www.robertlanfear.com/people/assets/rob_square.jpeg" /></td>
</tr>
</tbody>
</table>
</div>
<div id="section-1" class="section level2">
<h2>———————————————————————————————–</h2>
<hr />
</div>
<div class="section level2">
<h2>引言</h2>
<hr />
<blockquote>
<ul>
<li><h3 id="partitionfinder2"><a href="http://www.robertlanfear.com/partitionfinder/">PartitionFinder2</a>是目前对大中型数据集（数据类型包括：核苷酸、氨基酸和形态数据）同时检测最合适的分区方案和每个分区方案所对应的最优进化模型的最理想的程序（<strong>此观点来自布兰登·史塔克，如引用请注明出处</strong>）。</h3></li>
</ul>
</blockquote>
<blockquote>
<ul>
<li><h3 id="partitionfinder2-jmodeltest2protest3"><a href="http://www.robertlanfear.com/partitionfinder/">PartitionFinder2</a>所推演的最优进化模型结果与使用 <a href="https://code.google.com/p/jmodeltest2/">jModelTest2</a>（检测核苷酸数据）和<a href="http://code.google.com/p/prottest3/">ProTest3</a>（检测氨基酸数据）所推演的最优进化模型结果是比较接近的（<strong>此观点来自布兰登·史塔克，如引用请注明出处</strong>）。</h3></li>
</ul>
</blockquote>
<hr />
</div>
<div class="section level2">
<h2>简介</h2>
<hr />
<blockquote>
<ul>
<li><h3 id="-partitionfinder2">这次实战的目的只是介绍 <a href="http://www.robertlanfear.com/partitionfinder/">PartitionFinder2</a>最重要的功能。</h3></li>
</ul>
</blockquote>
<blockquote>
<ul>
<li><h3 id="-partitionfinder2">通过这次实战预期可以达到快速和高效的掌握 <a href="http://www.robertlanfear.com/partitionfinder/">PartitionFinder2</a>的一般使用方法。</h3></li>
</ul>
</blockquote>
<hr />
</div>
<div class="section level2">
<h2>态度</h2>
<hr />
<blockquote>
<ul>
<li><h3 id="partitionfinder2-rob-lanfearpartitionfinder2-2.1.x18"><a href="http://www.robertlanfear.com/partitionfinder/">PartitionFinder2</a>的作者 <a href="https://github.com/roblanf">Rob Lanfear</a>在<a href="http://www.robertlanfear.com/partitionfinder/">PartitionFinder2</a> <a href="http://www.robertlanfear.com/partitionfinder/assets/Manual_v2.1.x.pdf">2.1.x手册</a>（18页）里面提到：</h3></li>
</ul>
</blockquote>
<p><font color = red size = 4 face = "楷体">我目前最喜欢的用于进化模型选择的度量是AICc。一般来说，用户不应该用AIC，因为AICc与AIC相比始终是最好的。然而，AIC包含在partitionfinder里面大多是因为历史原因。</font></p>
<blockquote>
<ul>
<li><h3 id="partitionfinder2-rob-lanfearpartitionfinder2-2.1.x21"><a href="http://www.robertlanfear.com/partitionfinder/">PartitionFinder2</a>的作者 <a href="https://github.com/roblanf">Rob Lanfear</a>在<a href="http://www.robertlanfear.com/partitionfinder/">PartitionFinder2</a> <a href="http://www.robertlanfear.com/partitionfinder/assets/Manual_v2.1.x.pdf">2.1.x手册</a>（21页）里面还提到：</h3></li>
</ul>
</blockquote>
<p><font color = red size = 4 face = "楷体">一般来说，我不建议在任何情况下使用<code>search = hcluster;</code>这种算法。最好用<code>rcluster</code>这种算法并将<code>--rcluster-max</code>的参数设置为一些非常低的数值（例如，设置<code>--rcluster-max</code>为10）来代替。<code>hcluster</code>这种算法几乎与将<code>rcluster</code>算法的<code>--rcluster-max</code>参数设置为1是相同的。用户可以使用<code>--weights</code>命令行选项来控制<code>hcluster</code>算法。<code>hcluster</code>算法仍然放在<a href="http://www.robertlanfear.com/partitionfinder/">PartitionFinder2</a>里面纯粹是为了证明<code>hcluster</code>算法是不可重复使用的。</font></p>
<hr />
</div>
<div class="section level2">
<h2>安装</h2>
<hr />
<p><font color = green size = 5 face = "华文楷体">使用 <a href="http://continuum.io/downloads">Anaconda</a>或者其他程序安装
<a href="https://www.python.org/">Python</a></font></p>
<blockquote>
<ul>
<li><h4 id="partitionfinder2-python-2.7.10--python-3.x"><a href="http://www.robertlanfear.com/partitionfinder/">PartitionFinder2</a>需要 <a href="https://www.python.org/">Python</a> 2.7.10 或者更高版本才能运行，但是不是 <a href="https://www.python.org/">Python</a> 3.x版本。</h4></li>
<li><h4 id="anaconda-pythonhttpcontinuum.iodownloads">到目前为止，设置这个最简单的方法就是安装<a href="http://continuum.io/downloads">Anaconda Python</a>，这是一个简单的点击安装程序，可以在这里下载：<a href="http://continuum.io/downloads" class="uri">http://continuum.io/downloads</a>。</h4></li>
</ul>
</blockquote>
<blockquote>
<ul>
<li><h4 id="python-2.72.3.0anaconda-pythonanacondapython-2.7.xpython-2.7.10">遵循<a href="https://www.python.org/">Python</a> 2.7图形安装程序的链接，然后打开它并按照提示安装。用户需要确保有2.3.0版本或更高的版本<a href="http://continuum.io/downloads">Anaconda Python</a>。如果用户不想安装<a href="http://continuum.io/downloads">Anaconda</a>，可以安装用户自己喜欢的<a href="https://www.python.org/">Python</a> 2.7.x版本，但是需要不能低于Python 2.7.10版本，然后安装以下依赖包：</h4></li>
</ul>
</blockquote>
<blockquote>
<ul>
<li><font color = Fuchsia size = 4> <em>numpy</em></font></li>
</ul>
</blockquote>
<blockquote>
<ul>
<li><font color = Fuchsia size = 4> <em>pandas</em></font></li>
</ul>
</blockquote>
<blockquote>
<ul>
<li><font color = Fuchsia size = 4> <em>pytables</em></font></li>
</ul>
<ol style="list-style-type: decimal">
<li><h5 id="tablespytableshttpswww.pytables.orgusersguideinstallation.html">注意，安装的是tables包，而不是pytables(<a href="https://www.pytables.org/usersguide/installation.html)包" class="uri">https://www.pytables.org/usersguide/installation.html)包</a>。</h5></li>
<li><h5 id="pip-install-tables">安装的命令是：<code>pip install tables</code>。</h5></li>
<li><h5 id="pip-install---user---upgrade-tables">升级的命令是：<code>pip install --user --upgrade tables</code>。</h5></li>
</ol>
</blockquote>
<blockquote>
<ul>
<li><font color = Fuchsia size = 4> <em>pyparsing</em></font></li>
</ul>
</blockquote>
<blockquote>
<ul>
<li><font color = Fuchsia size = 4> <em>scipy</em></font></li>
</ul>
</blockquote>
<blockquote>
<ul>
<li><font color = Fuchsia size = 4> <em>sklearn</em></font></li>
</ul>
</blockquote>
<p><font color = green size = 5 face = "华文楷体">安装<a href="http://www.robertlanfear.com/partitionfinder/">PartitionFinder2</a></font></p>
<blockquote>
<ul>
<li><h4 id="partitionfinder2partitionfinder2">从<a href="http://www.robertlanfear.com/partitionfinder/">PartitionFinder2</a>官网下载最新版本的<a href="http://www.robertlanfear.com/partitionfinder/">PartitionFinder2</a>。</h4></li>
</ul>
</blockquote>
<blockquote>
<ul>
<li><h4 id=".zippartitionfinder2.1.1">双击以<code>.zip</code>为后缀的文件，它将自动解压。自动解压以后，用户可以得到一个文件夹，例如，<code>PartitionFinder2.1.1</code>。</h4></li>
</ul>
</blockquote>
<blockquote>
<ul>
<li><h4 id="partitionfinder2">把这个文件夹移动到用户想要存放<a href="http://www.robertlanfear.com/partitionfinder/">PartitionFinder2</a>的地方即可。</h4></li>
</ul>
</blockquote>
<hr />
</div>
<div class="section level2">
<h2>入门</h2>
<hr />
<p><font color = green size = 5 face = "华文楷体"><strong>序列数据</strong></font></p>
<blockquote>
<ul>
<li><h4 id="partitionfinder2phylippartitionfinder21-100"><a href="http://www.robertlanfear.com/partitionfinder/">PartitionFinder2</a>可以支持连续松弛型的PHYLIP格式的序列数据文件，<a href="http://www.robertlanfear.com/partitionfinder/">PartitionFinder2</a>中可以默认支持的序列名字符长度的范围是：<code>1-100</code>。</h4></li>
</ul>
</blockquote>
<p><font color = green size = 5 face = "华文楷体"> <strong>配置文件</strong></font></p>
<blockquote>
<ul>
<li><h4 id="partitionfinder2partition_finder.cfg"><a href="http://www.robertlanfear.com/partitionfinder/">PartitionFinder2</a>的配置文件为<code>partition_finder.cfg</code>，注意，该配置文件的文件名是固定不变的，请不要更改它。</h4></li>
</ul>
</blockquote>
<blockquote>
<ul>
<li><h4 id="partition_finder.cfg"><code>partition_finder.cfg</code>配置文件里面除了注释和方括号所在行的末尾不用分号结束，其他行的末尾要以分号结束。</h4></li>
</ul>
</blockquote>
<blockquote>
<ul>
<li><h4 id="partitionfinder2"><a href="http://www.robertlanfear.com/partitionfinder/">PartitionFinder2</a>的核苷酸配置文件举例如下：</h4></li>
</ul>
</blockquote>
<pre class="r"><code># ALIGNMENT FILE ##
alignment = test.phy;

## BRANCHLENGTHS: linked | unlinked ##
branchlengths = linked;

## MODELS OF EVOLUTION: all | allx | mrbayes | beast | gamma | gammai | &lt;list&gt; ##
models = GTR, GTR+G, GTR+I+G;

# MODEL SELECCTION: AIC | AICc | BIC #
model_selection = AICc;

## DATA BLOCKS: see manual for how to define ##
[data_blocks]
Gene1_pos1 = 1-789\3;
Gene1_pos2 = 2-789\3;
Gene1_pos3 = 3-789\3;
Gene2_pos1 = 790-1449\3;
Gene2_pos2 = 791-1449\3;
Gene2_pos3 = 792-1449\3;
Gene3_pos1 = 1450-2208\3;
Gene3_pos2 = 1451-2208\3;
Gene3_pos3 = 1452-2208\3;


## SCHEMES, search: all | user | greedy | rcluster | rclusterf | kmeans ##
[schemes]
search = greedy;</code></pre>
<hr />
<p><font color = green size = 5 face = "华文楷体"><strong>备注</strong></font></p>
<blockquote>
<ul>
<li><font color = Fuchsia size = 4> <em>alignment = test.phy</em>:</font><br />
</li>
</ul>
<ol style="list-style-type: decimal">
<li><h5 id="test.phy">序列数据文件名，本例中为<code>test.phy</code>。</h5></li>
<li><h5 id="partition_finder.cfgphylip">序列数据文件必须与<code>partition_finder.cfg</code>配置文件放在同一个文件夹里面，并且必须是正确的PHYLIP格式。</h5></li>
<li><h5 id="phyliptest.phytest.phy">PHYLIP格式的<code>test.phy</code>示例数据文件的信息如下（为了展示的效果，<code>test.phy</code>示例数据文件的后面部分的序列被截短了）：</h5></li>
</ol>
</blockquote>
<pre class="r"><code> 4 949
AD00P055  SLMLLISSSIVENGAGTGWTVYPPLSSNIAHSGSSVDLAIFSLHLAGISSILGAINFITTIINMKVNNLFFDQMSLFIWAVGITALLLLLSLPVLAGAITMLLTDRNLNT......
RV03N585  SLMLLISSSIVENGAGTGWTVYPPLSSNIAHSGSSVDLAIFSLHLAGISSILGAINFITTIINMKVNNLFFDQMSLFIWAVGITALLLLLSLPVLAGAITMLLTDRNLNT......
TDA99Q996 SLMLLISSSIVENGAGTGWTVYPPLSSNIAHSGSSVDLAIFSLHLAGISSILGAINFITTIINMKVNNMSFDQMSLFIWAVGITALLLLLSLPVLAGAITMLLTDRNLNT......
ZD99S305  SLMLLISSSIVENGAGTGWTVYPPLSSNIAHSGSSVDLAIFSLHLAGISSILGAINFITTIINMKVNNLFFDQMSLFIWAVGITALLLLLSLPVLAGAITMLLTDRNLNT......</code></pre>
<blockquote>
<ul>
<li><font color = Fuchsia size = 4> <em>user_tree_topology</em></font></li>
</ul>
<ol style="list-style-type: decimal">
<li><h5 id="alignmentpartition_finder.cfg">这是一个附加选项，可以在<code>alignment</code>行之后添加到<code>partition_finder.cfg</code>配置文件里面。</h5></li>
<li><h5 id="-partitionfinder2">如果用户想给 <a href="http://www.robertlanfear.com/partitionfinder/">PartitionFinder2</a>提供一个固定的拓扑结构树，而不是依靠软件默认估计的邻接法拓扑结构树，可以使用这个选项。</h5></li>
<li><h5>如果用户提前知道真正的树是什么，例如，在进行模拟时，这可能是有用的。</h5></li>
<li><h5 id="partition_finder.cfg">要使用该选项，只需在<code>partition_finder.cfg</code>配置文件里面额外添加一行即可，示例如下：</h5></li>
</ol>
</blockquote>
<pre class="r"><code>ALIGNMENT FILE #
alignment = test.phy;
user_tree_topology = tree.phy;</code></pre>
<blockquote>
<ol start="5" style="list-style-type: decimal">
<li><h5 id="tree.phynewicktree.phytest.phypartition_finder.cfg"><code>tree.phy</code>是一个包含<code>NEWICK</code>格式的拓扑学结构树（有或没有枝长）的文件名（不是路径）。文件名可以随意命名，它不必是<code>tree.phy</code>。树文件必须与序列数据文件（test.phy）和<code>partition_finder.cfg</code>配置文件在同一个文件夹中。</h5></li>
<li><h5>当用户使用此选项时，用户在树文件中提供的拓扑结构将在整个分析中是固定不变的。</h5></li>
<li><h5 id="gtrig">在标准分析中，使用GTR+I+G模型对整个数据集的枝长进行重新估计。</h5></li>
<li><h5 id="partition_finder.cfguser_tree_topology">如果用户不想使用这个选项，可以从<code>partition_finder.cfg</code>配置文件里面忽略带有<code>user_tree_topology</code>的这一行。</h5></li>
</ol>
</blockquote>
<blockquote>
<ul>
<li><font color = Fuchsia size = 4> <em>branchlengths = linked</em>:</font></li>
</ul>
<ol style="list-style-type: decimal">
<li><h5>这个选项决定在用户分析的过程中枝长是如何被估计的。用户如何设置这个选项在一定程度上取决于用户打算用哪一个程序进行最终的系统发育分析。</font></h5></li>
<li><h5 id="mrbayesbeastraxml">几乎所有的系统发育程序都支持相关联的枝长估计，但只有一些程序支持不相关联的枝长估计（例如，<a href="http://mrbayes.sourceforge.net/index.php">MrBayes</a>、<a href="http://beast.community/">BEAST</a>和<a href="https://sco.h-its.org/exelixis/web/software/raxml/">RAxML</a>）。</font></h5></li>
<li><h5 id="branchlengths-linked2n-3s-1ns5010106"><code>branchlengths = linked</code>的总枝长参数是<code>(2N-3)+(S-1)</code>，其中<code>N</code>代表物种数，<code>S</code>代表子集的数目。例如，如果有50个物种被分成10个子集，那么总枝长的参数就是106。</font></h5></li>
<li><h5 id="branchlengths-unlinked2ns-3sns5010970"><code>branchlengths = unlinked</code>的总枝长参数是<code>2NS-3S</code>,其中<code>N</code>代表物种数，<code>S</code>代表子集的数目。例如，如果有50个物种被分成10个子集，那么总枝长的参数就是970。</font></h5></li>
</ol>
</blockquote>
<blockquote>
<ul>
<li><font color = Fuchsia size = 4> <em>models = GTR, GTR+G, GTR+I+G</em>:</font></li>
</ul>
<ol style="list-style-type: decimal">
<li><h5>这个选项设置模型选择过程中需要考虑的进化模型。</h5></li>
<li><h5 id=".cfg">如果用户真想要所有可能模型的一个模型列表，可以在以.cfg为后缀的配置文件里面设置以下选项：</h5></li>
<li><h5 id="models-allx"><code>models = allx</code></h5></li>
<li><h5 id="models-allxf">这个模型参数在<code>models = all</code>的基础上还使用最大似然法（+X）而不是经验法（+F）对碱基或者氨基酸频率进行估计。</h5></li>
</ol>
</blockquote>
<blockquote>
<ul>
<li><font color = Fuchsia size = 4> <em>model_selection = aicc</em>:</font><br />
</li>
</ul>
<ol style="list-style-type: decimal">
<li><h5>设置用于模型选择的度量。</h5></li>
<li><h5 id="searchgreedy">如果用户使用<code>search=greedy</code>这个选项的话，它还定义了用于比较分区方案的度量。</h5></li>
</ol>
</blockquote>
<blockquote>
<ul>
<li><font color = Fuchsia size = 4> <em>[data_blocks]</em>:</font></li>
</ul>
<ol style="list-style-type: decimal">
<li><h5>设置数据分区。</h5></li>
<li><h5 id="partitionfinder2dnadnadna"><a href="http://www.robertlanfear.com/partitionfinder/">PartitionFinder2</a>可以处理三种类型数据：核苷酸、氨基酸和形态学数据。其中核苷酸数据又可以分为编码区DNA序列和非编码区DNA序列（例如，内含子）。编码区DNA序列根据密码子位置又进一步分为三个分区：第一位密码子、第二位密码子和第三位密码子。</h5></li>
<li><h5 id="partitionfinder2--partitionfinder2">根据<a href="http://www.robertlanfear.com/partitionfinder/">PartitionFinder2</a>软件作者的研究团队发表的 <a href="(http://mbe.oxfordjournals.org/content/32/6/1611)">文章</a>，他们推荐用户使用数据分区来告诉 <a href="http://www.robertlanfear.com/partitionfinder/">PartitionFinder2</a>尽可能多的关于用户序列的生物学信息。</h5></li>
<li><h5 id="dna2208bp789659bp758bp">设想一个DNA序列数据文件有2208bp的蛋白质编码区，分为三个基因，第一个基因的长度是789，第二个基因的长度是659bp,第三个基因的长度是758bp。那么就可以按照例子中的那样设置。</h5></li>
<li><h5 id="dna">氨基酸序列和非编码区DNA序列（例如，内含子）可以按照基因设置数据分区，提供基因的起始和终止位点即可。</h5></li>
<li><h5 id="k-means">对于形态学数据来说，通常没有直观的方法来定义数据分区。在这种情况下，用户可以尝试<code>k-means</code>算法。</h5></li>
</ol>
</blockquote>
<hr />
</div>
<div class="section level2">
<h2>实战</h2>
<hr />
<p><font color = green size = 5 face = "华文楷体"> <a href="http://www.robertlanfear.com/partitionfinder/">PartitionFinder2</a>的一般使用方法</font></p>
<hr />
<ul>
<li><font color = Fuchsia size = 4> <strong>对于一个小的多基因座数据集（例如，大约10基因座）</strong></font></li>
</ul>
<blockquote>
<ul>
<li>对于DNA序列，用<a href="http://www.atgc-montpellier.fr/phyml">PhyML</a>进行贪婪搜索。</li>
<li>对于氨基酸序列，用<a href="https://sco.h-its.org/exelixis/web/software/raxml/">RAxML</a>进行贪婪搜索。下面的这些选项分别给出了DNA和氨基酸序列的大多数模型。</li>
</ul>
<ol style="list-style-type: decimal">
<li><h5>根据基因和密码子位置定义数据分区</h5></li>
<li><h5 id=".cfg">在以.cfg为后缀的配置文件里面设置以下选项：</h5></li>
</ol>
</blockquote>
<pre class="r"><code>branchlengths = linked;
models = all;
model_selection = aicc;
search = greedy;</code></pre>
<blockquote>
<ol start="3" style="list-style-type: decimal">
<li><h5 id="dna-partitionfinder2">对于DNA数据集，从命令行运行 <a href="http://www.robertlanfear.com/partitionfinder/">PartitionFinder2</a>举例如下：</h5></li>
</ol>
</blockquote>
<pre class="r"><code>C:\Python27\python.exe C:\Users\BrandonStark\PartitionFinder\partitionfinder-2.1.1\PartitionFinder.py C:\Users\BrandonStark\PartitionFinder\partitionfinder-2.1.1\examples\nucleotide</code></pre>
<blockquote>
<ol start="4" style="list-style-type: decimal">
<li><h5 id="-partitionfinder2">对于氨基酸数据集，从命令行运行 <a href="http://www.robertlanfear.com/partitionfinder/">PartitionFinder2</a>举例如下：</h5></li>
</ol>
</blockquote>
<pre class="r"><code>C:\Python27\python.exe C:\Users\BrandonStark\PartitionFinder\partitionfinder-2.1.1\PartitionFinderProtein.py C:\Users\BrandonStark\PartitionFinder\partitionfinder-2.1.1\examples\aminoacid</code></pre>
<hr />
<ul>
<li><font color = Fuchsia size = 4> <strong>对于一个大的数据集（例如，大约100基因座）</strong></font></li>
</ul>
<blockquote>
<ul>
<li>对于DNA和氨基酸序列，用<a href="https://sco.h-its.org/exelixis/web/software/raxml/">RAxML</a>进行贪婪搜索。这通常相当快。</li>
<li>下面的这些选项分别给出了DNA和氨基酸序列的大多数模型。</li>
</ul>
<ol style="list-style-type: decimal">
<li><h5 id="-1">根据基因和密码子位置定义数据分区</h5></li>
<li><h5 id=".cfg-1">在以.cfg为后缀的配置文件里面设置以下选项：</h5></li>
</ol>
</blockquote>
<pre class="r"><code>branchlengths = linked;
models = all;
model_selection = AICc;
search = greedy;</code></pre>
<blockquote>
<ol start="3" style="list-style-type: decimal">
<li><h5 id="dna-raxml-partitionfinder2">对于DNA数据集，使用 <a href="https://sco.h-its.org/exelixis/web/software/raxml/">RAxML</a>从命令行运行 <a href="http://www.robertlanfear.com/partitionfinder/">PartitionFinder2</a>举例如下：</h5></li>
</ol>
</blockquote>
<pre class="r"><code>C:\Python27\python.exe C:\Users\BrandonStark\PartitionFinder\partitionfinder-2.1.1\PartitionFinder.py C:\Users\BrandonStark\PartitionFinder\partitionfinder-2.1.1\examples\nucleotide -- raxml</code></pre>
<blockquote>
<ol start="4" style="list-style-type: decimal">
<li><h5 id="-raxml-partitionfinder2">对于氨基酸数据集，使用 <a href="https://sco.h-its.org/exelixis/web/software/raxml/">RAxML</a>从命令行运行 <a href="http://www.robertlanfear.com/partitionfinder/">PartitionFinder2</a>举例如下：</h5></li>
</ol>
</blockquote>
<pre class="r"><code>C:\Python27\python.exe C:\Users\BrandonStark\PartitionFinder\partitionfinder-2.1.1\PartitionFinderProtein.py C:\Users\BrandonStark\PartitionFinder\partitionfinder-2.1.1\examples\aminoacid --raxml</code></pre>
<blockquote>
<ol start="5" style="list-style-type: decimal">
<li><h5 id="partition_finder.cfg">如果对于氨基酸数据集来说运行太慢的话，可能是因为有很多模型正在进行比较，因此尝试在<code>partition_finder.cfg</code>配置文件里面中指定一个较少的进化模型列表，例如：</h5></li>
</ol>
</blockquote>
<pre class="r"><code>models = LG, LG+G, LG+I+G, LG+I+G+F, LG4X;</code></pre>
<hr />
<ul>
<li><font color = Fuchsia size = 4> <strong>对于一个真正大的数据集（例如，大约1000基因座）</strong></font></li>
</ul>
<blockquote>
<ul>
<li>对于这种规模的数据集，贪婪算法很可能太慢。因此，我们可以使用一些更快的算法来尝试和猜测最好的分区方案。</li>
</ul>
<ol style="list-style-type: decimal">
<li><h5 id="-2">根据基因和密码子位置定义数据分区</h5></li>
<li><h5 id=".cfg-2">在以.cfg为后缀的配置文件里面设置以下选项：</h5>
<p>```R
branchlengths = linked;
models = all;
model_selection = aicc;
search = rcluster;</p></li>
<li><h5 id="dna-raxml-partitionfinder2-1">对于DNA数据集，使用 <a href="https://sco.h-its.org/exelixis/web/software/raxml/">RAxML</a>从命令行运行 <a href="http://www.robertlanfear.com/partitionfinder/">PartitionFinder2</a>举例如下：</h5></li>
</ol>
</blockquote>
<pre class="r"><code>C:\Python27\python.exe C:\Users\BrandonStark\PartitionFinder\partitionfinder-2.1.1\PartitionFinder.py C:\Users\BrandonStark\PartitionFinder\partitionfinder-2.1.1\examples\nucleotide -- raxml</code></pre>
<blockquote>
<ol start="4" style="list-style-type: decimal">
<li><h5 id="-raxml-partitionfinder2-1">对于氨基酸数据集，使用 <a href="https://sco.h-its.org/exelixis/web/software/raxml/">RAxML</a>从命令行运行 <a href="http://www.robertlanfear.com/partitionfinder/">PartitionFinder2</a>举例如下：</h5></li>
</ol>
</blockquote>
<pre class="r"><code>C:\Python27\python.exe C:\Users\BrandonStark\PartitionFinder\partitionfinder-2.1.1\PartitionFinderProtein.py C:\Users\BrandonStark\PartitionFinder\partitionfinder-2.1.1\examples\aminoacid -- raxml</code></pre>
<blockquote>
<ol start="5" style="list-style-type: decimal">
<li><h5 id="-partitionfinder2-rcluster-max-1000rcluster-max-100">这执行的是松弛型聚类算法,要想了解更多关于松弛型聚类算法的有关内容，用户可以浏览 <a href="http://www.robertlanfear.com/partitionfinder/">PartitionFinder2</a>作者的团队发表的 <a href="https://www.ncbi.nlm.nih.gov/pmc/articles/PMC4012149/">文章</a>。如果默认的命令对用户的数据集来说仍然太慢，请从默认的至少<code>rcluster-max = 1000</code>减少到例如<code>rcluster-max = 100</code>。关于这个的解释举例如下：</h5></li>
</ol>
</blockquote>
<pre class="r"><code>C:\Python27\python.exe C:\Users\BrandonStark\PartitionFinder\partitionfinder-2.1.1\PartitionFinderProtein.py C:\Users\BrandonStark\PartitionFinder\partitionfinder-2.1.1\examples\aminoacid -- raxml --rcluster-max 100</code></pre>
<blockquote>
<ol start="6" style="list-style-type: decimal">
<li><h5 id="rcluserrcluster-maxrcluster-percent"><code>rcluser</code>算法给用户在平衡速度（`–rcluster-max）和精度（–rcluster-percent）这两个参数之间有很多的控制权力。</h5></li>
</ol>
</blockquote>
<hr />
<ul>
<li><font color = Fuchsia size = 4> <strong>对于形态学数据集</strong></font></li>
</ul>
<blockquote>
<ul>
<li><p><a href="http://www.robertlanfear.com/partitionfinder/">PartitionFinder2</a>目前可以对形态学数据集进行四种不同的进化模型检测：
BINARY+G，BINARY+G+A，MULTISTATE+G，MULTISTATE+G+A。所有的这些模型都需要<code>--raxml</code>命令行选项，更多关于对形态学数据集进行操作的细节可以浏览的手册。</p></li>
<li><p>BINARY模型针对的是二进制数据集，MULTISTATE模型是MK模型针对的多状态数据集。</p></li>
<li><p>用户一次只可以用一个形态学模型分析数据集，这是由于二进制和多状态数据集的选择以及基于用户数据集本身的性质，需要提前决定是否需要为相应的数据集进行一个偏倚矫正（+A选项）。</p></li>
<li><p>对于一个给定的数据集，AIC/AICc/BIC方法对选择这些模型不是很合适。</p></li>
<li>形态学数据集<code>partition_finder.cfg</code>配置文件的设置，举例如下：</li>
</ul>
<ol style="list-style-type: decimal">
<li><h5>根据形态学数据集的长度定义单一数据分区</h5></li>
<li><h5 id=".cfg-3">在以.cfg为后缀的配置文件里面设置以下选项：</h5></li>
</ol>
</blockquote>
<pre class="r"><code>branchlengths = linked;
models = multistate+G;;
model_selection = aicc;
search = kmeans;</code></pre>
<blockquote>
<ol start="3" style="list-style-type: decimal">
<li><h5 id="-partitionfinder2">对于形态数据集，从命令行运行 <a href="http://www.robertlanfear.com/partitionfinder/">PartitionFinder2</a>举例如下：</h5></li>
</ol>
</blockquote>
<pre class="r"><code>C:\Python27\python.exe C:\Users\BrandonStark\PartitionFinder\partitionfinder-2.1.1\PartitionFinderProtein.py C:\Users\BrandonStark\PartitionFinder\partitionfinder-2.1.1\examples\morpholgy -- raxml</code></pre>
<hr />
<p><font color = green size = 5 face = "华文楷体"> 命令行选项</font></p>
<hr />
<blockquote>
<ul>
<li>还有一些附加的命令，用户可以从命令行将其传给 <a href="http://www.robertlanfear.com/partitionfinder/">PartitionFinder2</a>。这些命令可以用来微调用户的分析。</li>
</ul>
</blockquote>
<blockquote>
<ul>
<li><font color = Fuchsia size = 4><em>–all-states</em>:</font></li>
</ul>
<ol style="list-style-type: decimal">
<li><h5 id="k-means">只影响k-means算法。</h5></li>
<li><h5 id="k-means">具体来说，此选项限制k-means算法，只生成包含所有可能状态的子集。</h5></li>
</ol>
</blockquote>
<blockquote>
<ul>
<li><font color = Fuchsia size = 4><em>–force-restart</em>:</font></li>
</ul>
<ol style="list-style-type: decimal">
<li><h5 id="analysis">这将在重新启动运行之前删除以前的所有工作（删除`analysis文件夹）。</h5></li>
<li><h5 id="-partitionfinder2">默认情况下是不会这样做的，所以 <a href="http://www.robertlanfear.com/partitionfinder/">PartitionFinder2</a>可以使用已经计算的结果。</h5></li>
</ol>
</blockquote>
<blockquote>
<ul>
<li><font color = Fuchsia size = 4><em>–min-subset-size</em>:</font><br />
</li>
</ul>
<ol style="list-style-type: decimal">
<li><h4 id="100">默认：100。</h4></li>
<li><h4 id="k-meansk-meansmin-subset-size">只影响k-means算法。此选项限制k-means算法，只生成至少与<code>min-subset-size</code>一样大小的子集。</h4></li>
<li><h4 id="100k-means100">默认值是100个位点，所以默认情况下，k-means算法永远不会生成一个小于100个位点的子集。</h4></li>
</ol>
</blockquote>
<blockquote>
<ul>
<li><font color = Fuchsia size = 4> <em>–no-ml-tree</em>:</font><br />
</li>
</ul>
<ol style="list-style-type: decimal">
<li><h4 id="partitionfinder2"><a href="http://www.robertlanfear.com/partitionfinder/">PartitionFinder2</a>默认从用户的数据中估计一个最大似然树作为分析起始树。</h4></li>
<li><h4 id="--no-ml-treepartitionfinder2phyml-raxml">如果用户加上<code>--no-ml-tree</code>这个命令，那么<a href="http://www.robertlanfear.com/partitionfinder/">PartitionFinder2</a>将估计一个邻接法起始树（如果用户使用<a href="http://www.atgc-montpellier.fr/phyml">PhyML</a>）或者一个最大简约法起始树（如果用户使用 <a href="https://sco.h-its.org/exelixis/web/software/raxml/">RAxML</a>）。</h4></li>
</ol>
</blockquote>
<blockquote>
<ul>
<li><font color = Fuchsia size = 4> <em>–processors N, -p N</em>:</font></li>
</ul>
<ol style="list-style-type: decimal">
<li><h4>默认：使用所有可用的处理器。</h4></li>
<li><h5 id="n-partitionfinder2"><code>N</code>是用户想要 <a href="http://www.robertlanfear.com/partitionfinder/">PartitionFinder2</a>使用处理器的数目。</h5></li>
<li><h5 id="-partitionfinder2-4.--p-5partitionfinder25">但是，如果用户不希望 <a href="http://www.robertlanfear.com/partitionfinder/">PartitionFinder2</a>使用所有的处理器，就可以控制这个选项。 4. ##### 例如，设置<code>-p 5</code>将告诉<a href="http://www.robertlanfear.com/partitionfinder/">PartitionFinder2</a>一次最多使用5个处理器。</h5></li>
</ol>
</blockquote>
<blockquote>
<ul>
<li><font color = Fuchsia size = 4> <em>–quick, -q</em>:</font><br />
</li>
</ul>
<ol style="list-style-type: decimal">
<li><h4 id="partitionfinder2">此选项将阻止<a href="http://www.robertlanfear.com/partitionfinder/">PartitionFinder2</a>在分析的过程中写入不必要的关于分区方案的总结。</h4></li>
<li><h4>大多数人不需要使用这个选项，但是如果用户在进行真正的大数据分析，特别是使用贪婪算法的时候，它可以稍微加快速度。</h4></li>
</ol>
</blockquote>
<blockquote>
<ul>
<li><font color = Fuchsia size = 4> <em>– raxml</em>:</font></li>
</ul>
<ol style="list-style-type: decimal">
<li><h4 id="-partitionfinder2-raxmlphyml">这个命令告诉 <a href="http://www.robertlanfear.com/partitionfinder/">PartitionFinder2</a>使用 <a href="https://sco.h-its.org/exelixis/web/software/raxml/">RAxML</a>而不是<a href="http://www.atgc-montpellier.fr/phyml">PhyML</a>。</h4></li>
</ol>
</blockquote>
<blockquote>
<ul>
<li><font color = Fuchsia size = 4> <em>–rcluster-max N</em>:</font><br />
</li>
</ul>
<ol style="list-style-type: decimal">
<li><h4 id="1000">默认：1000。</h4></li>
<li><h4 id="--1">如果用户想要这个选项的数值无限大，请将其设置为<code>- 1</code>。</h4></li>
</ol>
</blockquote>
<blockquote>
<ul>
<li><font color = Fuchsia size = 4> <em>–rcluster-percent N</em>:</font></li>
</ul>
<ol style="list-style-type: decimal">
<li><h4 id="10">默认：10。</h4></li>
<li><h4 id="rcluster-maxrcluster-percent"><code>rcluster-max</code>和<code>rcluster-percent</code>一起控制松弛型聚类算法的彻底性。将其中任何一项的数值设置的更高一些往往会使搜索更彻底，但是会更慢。将他们的数值设置的更低一些往往会使搜索更快，但是会更不彻底。</h4></li>
</ol>
</blockquote>
<blockquote>
<ul>
<li><font color = Fuchsia size = 4> <em>–save-phylofiles</em>:</font></li>
</ul>
<ol style="list-style-type: decimal">
<li><h4 id="-partitionfinder2">这个选项将使 <a href="http://www.robertlanfear.com/partitionfinder/">PartitionFinder2</a>写入大量额外的文件到磁盘。</h4></li>
<li><h4>一般来说，用户不需要使用这个命令。</h4></li>
</ol>
</blockquote>
<blockquote>
<ul>
<li><font color = Fuchsia size = 4> <em>–weights “Wrate, Wbase, Wmodel, Walpha”</em>:</font></li>
</ul>
<ol style="list-style-type: decimal">
<li><h4 id="weights-1000">默认：–weights <code>1,0,0,0</code>。</h4></li>
<li><h4>在聚类算法中使用的权重列表。</h4></li>
<li><h4 id="alpha">该列表允许用户为一个子集的总替换速率、碱基/氨基酸频率、模型参数和<code>alpha</code>参数（它描述了位点之间的伽玛分布率）分配不同的权重。</h4></li>
</ol>
</blockquote>
<hr />
<p><font color = green size = 5 face = "华文楷体"> Windows系统下<a href="http://www.robertlanfear.com/partitionfinder/">PartitionFinder2</a>屏幕打印</font></p>
<hr />
<p>INFO | 2018-03-27 18:28:24,769 | config | Loading configuration at ’._finder.cfg’</p>
<p>INFO | 2018-03-27 18:28:24,778 | config | Setting ‘alignment’ to ‘test.phy’</p>
<p>INFO | 2018-03-27 18:28:24,778 | config | Setting ‘branchlengths’ to ‘linked’</p>
<p>INFO | 2018-03-27 18:28:24,779 | parser | Setting ‘models’ to a user-specified list</p>
<p>INFO | 2018-03-27 18:28:24,779 | parser | The models included in this analysis are: GTR+I+G, GTR, GTR+G</p>
<p>INFO | 2018-03-27 18:28:24,779 | parser | Setting datatype to ‘DNA’</p>
<p>INFO | 2018-03-27 18:28:24,779 | config | Setting ‘model_selection’ to ‘aicc’</p>
<p>INFO | 2018-03-27 18:28:24,785 | config | Setting ‘search’ to ‘greedy’</p>
<p>INFO | 2018-03-27 18:28:24,785 | config | ———————— BEGINNING NEW RUN ——————————-</p>
<p>INFO | 2018-03-27 18:28:24,846 | config | Looking for alignment file ‘..phy’…</p>
<p>INFO | 2018-03-27 18:28:24,848 | analysis | Beginning Analysis</p>
<p>INFO | 2018-03-27 18:28:24,848 | analysis | Removing Schemes in ‘.’ (they will be recalculated from existing subset data)</p>
<p>INFO | 2018-03-27 18:28:24,849 | config | Checking previously run configuration data…</p>
<p>INFO | 2018-03-27 18:28:24,861 | alignment | Reading alignment file ‘..phy’</p>
<p>INFO | 2018-03-27 18:28:24,894 | analysis | No starting tree file found.</p>
<p>INFO | 2018-03-27 18:28:24,897 | phyml | Making BioNJ tree for ._tree_source.phy</p>
<p>INFO | 2018-03-27 18:28:25,210 | phyml | Estimating GTR+I+G branch lengths on tree</p>
<p>INFO | 2018-03-27 18:28:26,124 | phyml | Branchlength estimation finished</p>
<p>INFO | 2018-03-27 18:28:26,125 | analysis | Starting tree with branch lengths is here: ._tree_source.phy_phyml_tree.txt</p>
<p>INFO | 2018-03-27 18:28:26,125 | method | Performing greedy analysis</p>
<p>INFO | 2018-03-27 18:28:26,125 | progress | PartitionFinder will have to analyse 73 subsets to complete this analysis</p>
<p>INFO | 2018-03-27 18:28:26,125 | progress | This will result in 121 schemes being created</p>
<p>INFO | 2018-03-27 18:28:26,125 | method | Analysing starting scheme (scheme start_scheme)</p>
<p>INFO | 2018-03-27 18:28:26,203 | threadpool | Found 4 cpus</p>
<p>INFO | 2018-03-27 18:28:26,599 | progress | Finished subset 1/73, 1.37 percent done</p>
<p>INFO | 2018-03-27 18:28:26,631 | progress | Finished subset 2/73, 2.74 percent done</p>
<p>INFO | 2018-03-27 18:28:26,776 | progress | Finished subset 3/73, 4.11 percent done</p>
<p>INFO | 2018-03-27 18:28:26,841 | progress | Finished subset 4/73, 5.48 percent done</p>
<p>INFO | 2018-03-27 18:28:27,115 | progress | Finished subset 5/73, 6.85 percent done</p>
<p>INFO | 2018-03-27 18:28:27,551 | progress | Finished subset 6/73, 8.22 percent done</p>
<p>INFO | 2018-03-27 18:28:27,867 | progress | Finished subset 7/73, 9.59 percent done</p>
<p>INFO | 2018-03-27 18:28:27,887 | progress | Finished subset 8/73, 10.96 percent done</p>
<p>INFO | 2018-03-27 18:28:27,917 | progress | Finished subset 9/73, 12.33 percent done</p>
<p>INFO | 2018-03-27 18:28:27,954 | method | <strong><em>Greedy algorithm step 1</em></strong></p>
<p>INFO | 2018-03-27 18:28:28,346 | progress | Finished subset 10/73, 13.70 percent done</p>
<p>INFO | 2018-03-27 18:28:29,096 | progress | Finished subset 11/73, 15.07 percent done</p>
<p>INFO | 2018-03-27 18:28:29,974 | progress | Finished subset 12/73, 16.44 percent done</p>
<p>INFO | 2018-03-27 18:28:30,288 | progress | Finished subset 13/73, 17.81 percent done</p>
<p>INFO | 2018-03-27 18:28:30,941 | progress | Finished subset 14/73, 19.18 percent done</p>
<p>INFO | 2018-03-27 18:28:31,548 | progress | Finished subset 15/73, 20.55 percent done</p>
<p>INFO | 2018-03-27 18:28:31,964 | progress | Finished subset 16/73, 21.92 percent done</p>
<p>INFO | 2018-03-27 18:28:32,760 | progress | Finished subset 17/73, 23.29 percent done</p>
<p>INFO | 2018-03-27 18:28:32,948 | progress | Finished subset 18/73, 24.66 percent done</p>
<p>INFO | 2018-03-27 18:28:33,280 | progress | Finished subset 19/73, 26.03 percent done</p>
<p>INFO | 2018-03-27 18:28:33,611 | progress | Finished subset 20/73, 27.40 percent done</p>
<p>INFO | 2018-03-27 18:28:33,887 | progress | Finished subset 21/73, 28.77 percent done</p>
<p>INFO | 2018-03-27 18:28:34,197 | progress | Finished subset 22/73, 30.14 percent done</p>
<p>INFO | 2018-03-27 18:28:34,400 | progress | Finished subset 23/73, 31.51 percent done</p>
<p>INFO | 2018-03-27 18:28:34,625 | progress | Finished subset 24/73, 32.88 percent done</p>
<p>INFO | 2018-03-27 18:28:35,141 | progress | Finished subset 25/73, 34.25 percent done</p>
<p>INFO | 2018-03-27 18:28:35,437 | progress | Finished subset 26/73, 35.62 percent done</p>
<p>INFO | 2018-03-27 18:28:35,684 | progress | Finished subset 27/73, 36.99 percent done</p>
<p>INFO | 2018-03-27 18:28:35,983 | progress | Finished subset 28/73, 38.36 percent done</p>
<p>INFO | 2018-03-27 18:28:36,180 | progress | Finished subset 29/73, 39.73 percent done</p>
<p>INFO | 2018-03-27 18:28:36,400 | progress | Finished subset 30/73, 41.10 percent done</p>
<p>INFO | 2018-03-27 18:28:36,736 | progress | Finished subset 31/73, 42.47 percent done</p>
<p>INFO | 2018-03-27 18:28:37,865 | progress | Finished subset 32/73, 43.84 percent done</p>
<p>INFO | 2018-03-27 18:28:38,278 | progress | Finished subset 33/73, 45.21 percent done</p>
<p>INFO | 2018-03-27 18:28:38,640 | progress | Finished subset 34/73, 46.58 percent done</p>
<p>INFO | 2018-03-27 18:28:39,214 | progress | Finished subset 35/73, 47.95 percent done</p>
<p>INFO | 2018-03-27 18:28:39,459 | progress | Finished subset 36/73, 49.32 percent done</p>
<p>INFO | 2018-03-27 18:28:39,742 | progress | Finished subset 37/73, 50.68 percent done</p>
<p>INFO | 2018-03-27 18:28:39,940 | progress | Finished subset 38/73, 52.05 percent done</p>
<p>INFO | 2018-03-27 18:28:40,118 | progress | Finished subset 39/73, 53.42 percent done</p>
<p>INFO | 2018-03-27 18:28:40,450 | progress | Finished subset 40/73, 54.79 percent done</p>
<p>INFO | 2018-03-27 18:28:41,061 | progress | Finished subset 41/73, 56.16 percent done</p>
<p>INFO | 2018-03-27 18:28:41,256 | progress | Finished subset 42/73, 57.53 percent done</p>
<p>INFO | 2018-03-27 18:28:41,654 | progress | Finished subset 43/73, 58.90 percent done</p>
<p>INFO | 2018-03-27 18:28:41,924 | progress | Finished subset 44/73, 60.27 percent done</p>
<p>INFO | 2018-03-27 18:28:42,134 | progress | Finished subset 45/73, 61.64 percent done</p>
<p>INFO | 2018-03-27 18:28:42,183 | method | <strong><em>Greedy algorithm step 2</em></strong></p>
<p>INFO | 2018-03-27 18:28:42,565 | progress | Finished subset 46/73, 63.01 percent done</p>
<p>INFO | 2018-03-27 18:28:42,842 | progress | Finished subset 47/73, 64.38 percent done</p>
<p>INFO | 2018-03-27 18:28:43,275 | progress | Finished subset 48/73, 65.75 percent done</p>
<p>INFO | 2018-03-27 18:28:43,585 | progress | Finished subset 49/73, 67.12 percent done</p>
<p>INFO | 2018-03-27 18:28:43,816 | progress | Finished subset 50/73, 68.49 percent done</p>
<p>INFO | 2018-03-27 18:28:44,132 | progress | Finished subset 51/73, 69.86 percent done</p>
<p>INFO | 2018-03-27 18:28:44,322 | progress | Finished subset 52/73, 71.23 percent done</p>
<p>INFO | 2018-03-27 18:28:44,339 | method | <strong><em>Greedy algorithm step 3</em></strong></p>
<p>INFO | 2018-03-27 18:28:44,967 | progress | Finished subset 53/73, 72.60 percent done</p>
<p>INFO | 2018-03-27 18:28:45,645 | progress | Finished subset 54/73, 73.97 percent done</p>
<p>INFO | 2018-03-27 18:28:46,799 | progress | Finished subset 55/73, 75.34 percent done</p>
<p>INFO | 2018-03-27 18:28:47,676 | progress | Finished subset 56/73, 76.71 percent done</p>
<p>INFO | 2018-03-27 18:28:48,267 | progress | Finished subset 57/73, 78.08 percent done</p>
<p>INFO | 2018-03-27 18:28:49,032 | progress | Finished subset 58/73, 79.45 percent done</p>
<p>INFO | 2018-03-27 18:28:49,144 | method | <strong><em>Greedy algorithm step 4</em></strong></p>
<p>INFO | 2018-03-27 18:28:49,759 | progress | Finished subset 59/73, 80.82 percent done</p>
<p>INFO | 2018-03-27 18:28:49,953 | progress | Finished subset 60/73, 82.19 percent done</p>
<p>INFO | 2018-03-27 18:28:50,190 | progress | Finished subset 61/73, 83.56 percent done</p>
<p>INFO | 2018-03-27 18:28:50,483 | progress | Finished subset 62/73, 84.93 percent done</p>
<p>INFO | 2018-03-27 18:28:50,801 | progress | Finished subset 63/73, 86.30 percent done</p>
<p>INFO | 2018-03-27 18:28:50,861 | method | <strong><em>Greedy algorithm step 5</em></strong></p>
<p>INFO | 2018-03-27 18:28:51,947 | progress | Finished subset 64/73, 87.67 percent done</p>
<p>INFO | 2018-03-27 18:28:52,180 | progress | Finished subset 65/73, 89.04 percent done</p>
<p>INFO | 2018-03-27 18:28:52,483 | progress | Finished subset 66/73, 90.41 percent done</p>
<p>INFO | 2018-03-27 18:28:52,688 | progress | Finished subset 67/73, 91.78 percent done</p>
<p>INFO | 2018-03-27 18:28:52,698 | method | Greedy algorithm finished after 5 steps</p>
<p>INFO | 2018-03-27 18:28:52,700 | method | Highest scoring scheme is scheme step_4, with aicc score of 10935.186</p>
<p>INFO | 2018-03-27 18:28:52,700 | reporter | Information on best scheme is here: ._scheme.txt</p>
<p>INFO | 2018-03-27 18:28:52,703 | main | Total processing time: 0:00:28 (h:m:s)</p>
<p>INFO | 2018-03-27 18:28:52,703 | main | Processing complete.</p>
<hr />
<p><font color = green size = 5 face = "华文楷体"> Windows系统下<a href="http://www.robertlanfear.com/partitionfinder/">PartitionFinder2</a>文件输出</font></p>
<hr />
<blockquote>
<ul>
<li>根据以上指令，<a href="http://www.robertlanfear.com/partitionfinder/">PartitionFinder2</a>运行完成后，所有的输出都在名叫<code>analysis</code>的一个文件夹里面，它与用户的序列数据文件在同一个文件夹里面。</li>
</ul>
</blockquote>
<blockquote>
<ul>
<li>在<code>analysis</code>这个文件夹里面有很多的输出，但是，一般情况下，用户可能只对其中的三个文件或者文件夹感兴趣，它们的顺序可能是：</li>
</ul>
</blockquote>
<blockquote>
<ul>
<li><font color = Fuchsia size = 4> <em>best_schemes.txt</em>:</font></li>
</ul>
<ol style="list-style-type: decimal">
<li><h4>这个最佳分区方案文件有关于发现最佳分区方案的信息，以及用于找到最佳分区方案的设置。</h4></li>
<li><h4 id="-raxmlnexus">这包括该方案的详细描述，以及该方案中每个子集选择的分子进化模型。它还包含每个方案在 <a href="https://sco.h-its.org/exelixis/web/software/raxml/">RAxML</a>和NEXUS格式的一个描述。</h4></li>
</ol>
</blockquote>
<blockquote>
<ul>
<li><font color = Fuchsia size = 4> <em>subsets folder</em>：</font></li>
</ul>
<ol style="list-style-type: decimal">
<li><h4>这个子集文件夹包含每个子集模型选择的分析结果。</h4></li>
<li><h4 id=".txtaicc">它们都是以<code>.txt</code>为扩展名的文件，其中，用户纳入分析的每个模型以AICc增序的方式列成表（即最好的模型在顶部）。</h4></li>
</ol>
</blockquote>
<blockquote>
<ul>
<li><font color = Fuchsia size = 4> <em>schemes folder</em>：</font></li>
</ul>
<ol style="list-style-type: decimal">
<li><h4>这个方案文件夹包含分析方案的详细信息。</h4></li>
<li><h4 id=".txtbest_scheme.txt">每一个方案分别在一个以<code>.txt</code>为扩展名的独立文件里面，很像<code>best_scheme.txt</code>文件。</h4></li>
</ol>
</blockquote>
<hr />
</div>
<div class="section level2">
<h2>引用</h2>
<hr />
<p><font color = green size = 5 face = "华文楷体">根据用户的分析，可能需要引用多达三篇文献：</font></p>
<ul>
<li><font color = Fuchsia size = 4> <strong>一个是关于<a href="http://www.robertlanfear.com/partitionfinder/">PartitionFinder2</a>程序</strong></font></li>
</ul>
<blockquote>
<ul>
<li>如果您在任何出版作品中使用此程序，请引用:</li>
</ul>
</blockquote>
<pre class="r"><code>Lanfear, R., Frandsen, P. B., Wright, A. M., Senfeld, T., Calcott, B. (2016)
PartitionFinder 2: new methods for selecting partitioned models of evolution for
molecular and morphological phylogenetic analyses. Molecular biology and evolution.
DOI: dx.doi.org/10.1093/molbev/msw260.</code></pre>
<ul>
<li><font color = Fuchsia size = 4> <strong>一个是用户所使用的算法</strong></font></li>
</ul>
<blockquote>
<ol style="list-style-type: decimal">
<li>如果用户使用<code>search = rcluster</code>或者<code>search = hcluster</code>，请引用：</li>
</ol>
</blockquote>
<pre class="r"><code>Lanfear, R., Calcott, B., Kainer, D., Mayer, C., &amp; Stamatakis, A. (2014). Selecting
optimal partitioning schemes for phylogenomic datasets. BMC evolutionary
biology, 14(1), 82.</code></pre>
<blockquote>
<ol start="2" style="list-style-type: decimal">
<li>如果用户使用<code>search = kmeans</code>，请引用：</li>
</ol>
</blockquote>
<pre class="r"><code>Frandsen, P. B., Calcott, B., Mayer, C., &amp; Lanfear, R. (2015). Automatic selection of
partitioning schemes for phylogenetic analyses using iterative k-means clustering of site
rates. BMC Evolutionary Biology, 15(1), 13.</code></pre>
<ul>
<li><font color = Fuchsia size = 4> <strong>一个是关于使用<a href="http://www.atgc-montpellier.fr/phyml">PhyML</a>或者<a href="https://sco.h-its.org/exelixis/web/software/raxml/">RAxML</a></strong></font></li>
</ul>
<blockquote>
<ol style="list-style-type: decimal">
<li>如果你使用 <a href="http://www.robertlanfear.com/partitionfinder/">PartitionFinder2</a>没有<code>——RAxML</code>命令行选项，<a href="http://www.robertlanfear.com/partitionfinder/">PartitionFinder2</a>在很大程度上依赖，用<a href="http://www.atgc-montpellier.fr/phyml">PhyML</a> 3.0 版本，所以请引用：</li>
</ol>
</blockquote>
<pre class="r"><code>New Algorithms and Methods to Estimate Maximum-Likelihood Phylogenies: Assessing
the Performance of PhyML 3.0. Guindon S., Dufayard J.F., Lefort V., Anisimova M.,
Hordijk W., Gascuel O. Systematic Biology, 59(3):307-21, 2010.</code></pre>
<blockquote>
<ol start="2" style="list-style-type: decimal">
<li>如果用户使用<code>-- raxml</code>命令行选项，<a href="http://www.robertlanfear.com/partitionfinder/">PartitionFinder2</a>使用<a href="https://sco.h-its.org/exelixis/web/software/raxml/">RAxML</a> 8.0版本用于计算。如果用户使用这个命令行选项，请引用：</li>
</ol>
</blockquote>
<pre class="r"><code>A. Stamatakis, RAxML-VI-HPC: maximum likelihood-based phylogenetic analyses with
thousands of taxa and mixed models, Bioinformatics 22, 2688–2690 (2006).</code></pre>
<hr />
</div>
<div class="section level2">
<h2>英文网站</h2>
<hr />
<p><font color = green size = 5 face = "华文楷体">想了解更多关于 <a href="http://www.robertlanfear.com/partitionfinder/">PartitionFinder2</a>有关的内容请浏览 <a href="http://www.robertlanfear.com/partitionfinder/">PartitionFinder2</a>的官方网站 <a href="http://www.robertlanfear.com/partitionfinder/">PartitionFinder2</a>。</font></p>
<hr />
</div>
<div class="section level2">
<h2>联系我们</h2>
<hr />
<p><font color = green size = 5 face = "华文楷体">如果您有任何关于<a href="http://www.robertlanfear.com/partitionfinder/">PartitionFinder2</a>相关的问题，欢迎您到我们的计算进化交流群（QQ: 153258468）与我们一起探讨。</font></p>
</div>
<div id="section-2" class="section level2">
<h2>———————————————————————————————–</h2>
<hr />
<div id="jon-snow" class="section level3">
<h3><strong>Jon Snow</strong></h3>
<p><img src="https://gss0.bdstatic.com/-4o3dSag_xI4khGkpoWK1HF6hhy/baike/w%3D268%3Bg%3D0/sign=23437e3cd333c895a67e9f7de92814cd/b3b7d0a20cf431ade7884a924136acaf2fdd98c6.jpg" /></p>
</div>
<div id="email-376063966qq.com" class="section level3">
<h3>email: <a href="mailto:376063966@qq.com">376063966@qq.com</a></h3>
</div>
<div id="research-interests" class="section level3">
<h3><font color = Aqua size = 5><em>Research Interests</em>:</font></h3>
<blockquote>
<ul>
<li>I am broadly interested in bioinformatics, with particular focus on the influenza virus.</li>
</ul>
</blockquote>
</div>
</div>
<div id="section-3" class="section level2">
<h2>———————————————————————————————–</h2>
<hr />
<div id="brandon-stark" class="section level3">
<h3><strong>Brandon Stark</strong></h3>
<p><img src="https://gss2.bdstatic.com/9fo3dSag_xI4khGkpoWK1HF6hhy/baike/w%3D268%3Bg%3D0/sign=1e963699880a19d8cb0383030bc1e5b6/503d269759ee3d6d87d2e8e34b166d224f4ade86.jpg" /></p>
</div>
<div id="email-2049968646qq.com" class="section level3">
<h3>email: <a href="mailto:2049968646@qq.com">2049968646@qq.com</a></h3>
</div>
<div id="research-interests-1" class="section level3">
<h3><font color = Aqua size = 5><em>Research Interests</em>:</font></h3>
<blockquote>
<ul>
<li>better understanding the evolution of emerging human viral pathogens, in particular, fast evolving RNA viruses that have been sampled through time;</li>
<li>Inferring evolutionary and population dynamic processes from molecular sequence data;</li>
<li>Models of molecular evolution for phylogenetic reconstruction using maximum likelihood and Bayesian inference.</li>
<li>I am also interested in empirical and theoretical approaches focusing on the use of experimental evolutionary, population genetic, phylogenetic, and computational methods.</font></li>
</ul>
</blockquote>
</div>
</div>
<div id="section-4" class="section level2">
<h2>———————————————————————————————–</h2>
<hr />
</div>
